File: 1448.CountGoodNodesinBinaryTree.go
func goodNodes(root *TreeNode) int {
    return recur(root, math.MinInt)
}

func recur(root *TreeNode, curMax int) int{
    if root == nil{
        return 0
    }

    res := 0
    

    if root.Val >= curMax{
        res++
        curMax = root.Val
    }

    res += recur(root.Left, curMax)
    res += recur(root.Right, curMax)

    return res
}

File: 1448.CountGoodNodesinBinaryTree.py
class Solution:
    def __init__(self):
        self.cnt = 0
    def goodNodes(self, root: TreeNode) -> int:
        def recur(node, curMax):
            if not node:
                return
            if node.val >= curMax:
                curMax = node.val
                self.cnt += 1
            
            recur(node.left, curMax)
            recur(node.right, curMax)

        recur(root, float("-inf"))
        return self.cnt
